exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -1010,7 +1010,9 @@ public void addToEventQueue(final QueryState newState, final Exception exception
 
   private void recordNewState(final QueryState newState) {
     state = newState;
-    queryManager.updateEphemeralState(newState);
+    if (queryContext.getOptions().getOption(ExecConstants.ZK_QUERY_STATE_UPDATE)) {
+      queryManager.updateEphemeralState(newState);
+    }
```

Comment:
How does this affect query operation for long-running queries? How does it impact the query profile? If updates are enabled, do we still do an update at query completion to finalize the profile? If not, should writing of the profile be automatically disabled if status updates are disabled?

Do we do any timeout on updates? Will we notice that the query has not been updated and, say, kill the query due to timeouts?